plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java

```diff
@@ -173,44 +170,48 @@ void setTransport(String transport) {
     this.transport = transport;
   }
   
-  public synchronized boolean isResolved() {
-    return this.resolved;
+  /**
+   * Direct dependencies only. Do not attempt to load dependencies from Maven Central.
+   */
+  synchronized void setResolved() {
+    if (transitiveDependencies == null) {
+      transitiveDependencies = directDependencies;
+    }
   }
-  
+
   /**
-   * @param resolved true iff this library contains its complete dependency graph
+   * Returns the non-transitive dependencies of this library. Useful when a separate system such as
+   * Maven will resolve the transitive dependencies later. 
    */
-  synchronized void setResolved(boolean resolved) {
-    this.resolved = resolved;
+  public List<LibraryFile> getDirectDependencies() {
+    return new ArrayList<>(this.directDependencies);
```

Comment:
`ImmutableList`?

Reply:
Not a bad idea. Right now I'd prefer not to make further changes.